### PR TITLE
feat(popup): allow short form for wallet addresses in popup

### LINKED ISSUE #411: Users should have the ability to use the short and a long form for a payment pointer/wallet address
### Context

- Users should be able to use the short form (`$`) and the long form (`$`) - ONLY in the extension's UI

### Todos

- Convert `$` to `https:`

### PR DESCRIPTION
**Pull request by raducristianpopa**
## Context

Closes #411.

## Changes proposed in this pull request

Discussed during WM Work Week. ASEs can decide what term they want to use - wallet address or payment pointer. This PR allows the use of the short form ($ notation) in the popup.

### CODE REVIEW

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">152cf9897577af50294318c46407eb805cd92389</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/9925747901">Run #9925747901</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/25979375348/artifacts/1698920695">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/25979375348/artifacts/1698920697">Download</a></td></tr>
</tbody>
</table>

File: src/popup/lib/utils.ts [hunk 1]
```diff
@@ -62,3 +62,7 @@ export function formatNumber(
     } else return value.toExponential()
   }
 }
+
+export function toWalletAddressUrl(s: string): string {
+  return s.startsWith('$') ? s.replace(/^\$/, 'https://') : s
```
Comment (sidvishnoi) [inline src/popup/lib/utils.ts:67]: Add a test just to be sure?

File: src/popup/components/ConnectWalletForm.tsx [hunk 2]
```diff
@@ -137,7 +142,7 @@ export const ConnectWalletForm = ({ publicKey }: ConnectWalletFormProps) => {
         <Code className="text-xs" value={publicKey} />
       </div>
       <Input
-        type="url"
+        type="text"
         label="Wallet address"
```
Comment (sidvishnoi) [inline src/popup/components/ConnectWalletForm.tsx:146]: Should we say: "Wallet address or payment pointer"?

File: src/popup/components/ConnectWalletForm.tsx [hunk 3]
```diff
@@ -52,7 +53,8 @@ export const ConnectWalletForm = ({ publicKey }: ConnectWalletFormProps) => {
       clearErrors('walletAddressUrl')
       if (!walletAddressUrl) return
       try {
-        const url = new URL(walletAddressUrl)
+        const httpUrl = toWalletAddressUrl(walletAddressUrl)
+        const url = new URL(httpUrl)
```
Comment (sidvishnoi) [inline src/popup/components/ConnectWalletForm.tsx:57]:
```suggestion
        const url = new URL(toWalletAddressUrl(walletAddressUrl))
```

File: src/popup/lib/utils.ts (hunk 1 above)
Comment (sidvishnoi) [inline src/popup/lib/utils.ts:67]: As it'll only replace first one and we tested it starts with `$`
```suggestion
  return s.startsWith('$') ? s.replace('$', 'https://') : s
```

**Review by sidvishnoi — APPROVED** [review]

**Pull request merged**